src/io.rs

```diff
@@ -67,19 +71,23 @@ impl<R: Read> Scanner<R> {
     }
 
     pub fn next_line(&mut self) -> String {
-        assert!(self.tokens.is_empty(), "You have unprocessed token");
+        assert!(self.tokens.is_empty(), "You have unprocessed token(s)");
         self.lines
             .next()
             .and_then(|e| e.ok())
-            .expect("Failed to read.")
+            .expect("next_line failed.")
+    }
+
+    pub fn is_empty(&mut self) -> bool {
```

Comment:
Could use ```has_next``` like in Java. Otherwise, I can use ```at_line_start```, but I think this is also unclear.
The function returns true if and only if this scanner has another token, so a name like ```has_tokens``` could also work.
Obviously, the function would need to instead return the result of ```!self.tokens.is_empty```.